File: CPE-1star/UVA299 -Train Swapping.py
#氣泡排序，由小到大，計算交換的次數
def bbsort(subnum):
	count=0
	for k in range(len(subnum)-1):
		for j in range(0,len(subnum)-1):
			if subnum[j]>subnum[j+1]:
				temp=subnum[j+1]
				subnum[j+1]=subnum[j]
				subnum[j]=temp
				count+=1
	return count
	
while True:
	try:
		n=int(input())
		for i in range(n):
			l=int(input())
			nums=list(map(int,input().split()))
			print("Optimal train swapping takes %d swaps."%(bbsort(nums)))
	except EOFError:
		break

File: CPE-1star/UVA299 -Train Swapping.cpp
#include<iostream>
#include <vector>
using namespace std;
int bbsort(vector<int>& subnum){
    int count=0;
    for(int k=0;k<subnum.size()-1;k++)
        for(int j=0;j<subnum.size()-1;j++){
            if(subnum[j]>subnum[j+1]){
                int temp=subnum[j+1];
                subnum[j+1]=subnum[j];
                subnum[j]=temp;
                count+=1;
            }
        }
    return count;
}
int main(){
    int n;
    while(cin>>n){
        for(int i=0;i<n;i++){
            int l;
            cin>>l;
            vector<int> nums(l);
            for(int j=0;j<l;j++){
                int num;
                cin>>num;
                nums[j]=num;
            }
            cout<<"Optimal train swapping takes "<<bbsort(nums)<<" swaps."<<"\n";
        }
    }
    return 0;
}